src/components/ArticleList.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from "react";
+import { fetchArticles } from "./api";
+import ArticleCard from "./ArticleCard.jsx";
+
+export default function ArticleList() {
+  const [articleList, setArticleList] = useState([]);
+  const [limit, setLimit] = useState(10);
+
+  //increase number of articles shown when button clicked
+  const incrementLimit = async (increment) => {
+    setLimit((currLimit) => {
+      return currLimit + increment;
+    });
+  };
+
+  const loadArticles = async () => {
+    const articles = await fetchArticles(limit);
+    setArticleList(articles);
+  };
+
+  // articles will re-render each time limit is amended
+  useEffect(() => {
+    loadArticles();
+  }, [limit]);
+
+  const IncrementButton = () => {
+    if (articleList[0] && limit > articleList[0].total_count)
+      return <h2>No more articles</h2>;
+    return (
+      <button
+        onClick={() => {
+          incrementLimit(5);
+        }}
+      >
+        Load more Articles
+      </button>
+    );
+  };
```

Comment:
As this is its own component it should be extracted out to its own file.

Reply:
Thanks Carrie - i've moved it out of the file now:)